tests/rptest/tests/transactions_test.py

```diff
@@ -127,17 +130,20 @@ def simple_test(self):
         })
         consumer2.subscribe([self.output_t])
 
-        final_consume_cnt = 0
+        index_from_input = 0
         expected = bytes("0", 'UTF-8')
 
-        while final_consume_cnt != self.max_records:
+        while index_from_input < self.max_records:
             records = self.consume(consumer2)
 
             for record in records:
-                assert (record.key() == expected)
```

Comment:
You know why the test failed in the first place? I didn't see much info in the commit/related issue.

Reply:
I do not know
Just rewrite check and add logging